### PR TITLE
perf(django): reimplement django simple with concurrency

### PR DESCRIPTION
**Pull request by majorgreys**
## Description

This reimplements the Django simple application as a single file to reduce the complexity of building the application and dependencies. Each request produces ~25 spans in a trace.

Also, added here is the case of concurrent requests to both a single gunicorn worker with 1 thread to a single gunicorn worker with 10 threads.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: benchmarks/django_simple/README.rst [hunk 1]
```diff
@@ -8,25 +8,13 @@ Build
...
-This image will by default install the release version of ``ddtrace``.
-
-You can install a different version by using git tags or commit hashes as the value for the ``DDTRACE_GIT_COMMIT_ID`` environment variable::
-
-  docker run -it --rm -e "DDTRACE_GIT_COMMIT_ID=v0.48.1" django_simple
-
-You can also install from a set of wheel by mounting a volume with the necessary wheels and setting the ``DDTRACE_WHEELS`` environment variable::
-
-  docker run -it --rm  -e "DDTRACE_WHEELS=/artifacts/wheels" -v "/path/to/wheels/":"/artifacts/wheels/" django_simple
+If you want to save the output, mount a volume to ``/artifacts/``.
```
Comment (Kyle-Verhoog) [inline benchmarks/django_simple/README.rst:20]: I'd be in favour of leaving in an example for this (mostly bc I always forget the syntax for specifying volumes). Might be worth including in the above command for an easy copy-paste one-liner.

File: benchmarks/django_simple/gunicorn.conf.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def post_worker_init(worker):
+    # If profiling enabled but not tracer than only run auto script for profiler
+    if os.environ.get("PERF_ENABLE_PROFILER") == "1" and os.environ.get("PERF_ENABLE_TRACER") == "0":
+        import ddtrace.profiling.auto  # noqa
+
+
+bind = "0.0.0.0:8000"
+if os.environ.get("PERF_THREADS"):
+    worker_class = "gthread"
+    threads = 10
```
Comment (Kyle-Verhoog) [inline benchmarks/django_simple/gunicorn.conf.py:25]: worth making num threads configurable too? maybe just reuse `PERF_THREADS` and treat > 0 as number of threads?

**Review by Kyle-Verhoog — COMMENTED** [review]
Nice! A lot better without the billion Django files - especially for our use-case :+1:

**Comment by Kyle-Verhoog** [comment]
@majorgreys it'd be cool to show some of the results that this scenario produces here in the PR just for interest's sake

File: benchmarks/django_simple/scenario.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+VARIANTS = [
+    dict(PERF_ENABLE_TRACER="0", PERF_ENABLE_PROFILER="0", PERF_THREADS="0"),
+    dict(PERF_ENABLE_TRACER="1", PERF_ENABLE_PROFILER="0", PERF_THREADS="0"),
+    dict(PERF_ENABLE_TRACER="0", PERF_ENABLE_PROFILER="1", PERF_THREADS="0"),
+    dict(PERF_ENABLE_TRACER="1", PERF_ENABLE_PROFILER="1", PERF_THREADS="0"),
+    dict(PERF_ENABLE_TRACER="0", PERF_ENABLE_PROFILER="0", PERF_THREADS="1"),
+    dict(PERF_ENABLE_TRACER="1", PERF_ENABLE_PROFILER="0", PERF_THREADS="1"),
+    dict(PERF_ENABLE_TRACER="0", PERF_ENABLE_PROFILER="1", PERF_THREADS="1"),
+    dict(PERF_ENABLE_TRACER="1", PERF_ENABLE_PROFILER="1", PERF_THREADS="1"),
+]
```
Comment (P403n1x87) [inline benchmarks/django_simple/scenario.py:24]:
```suggestion
from itertools import product 


VARIABLES = ("PERF_ENABLE_TRACER", "PERF_ENABLE_PROFILER", "PERF_THREADS")
VARIANTS = [
    dict(zip(VARIABLES, _)) for _ in product(*[[0,1]] * len(VARIABLES))
]
```
just a suggestion

**Review by P403n1x87 — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**